xonsh/ptk_shell/shell.py

```diff
@@ -351,6 +352,8 @@ def singleline(
             for attr, val in self.get_lazy_ptk_kwargs():
                 prompt_args[attr] = val
 
+        if editing_mode == EditingMode.VI:
+            prompt_args["cursor"] = ModalCursorShapeConfig()
```

Comment:
I see that this is a new feature since 3.0.27 
https://github.com/prompt-toolkit/python-prompt-toolkit/blob/51ccb92d07e9a1a2138519cf03059bd806d9c487/CHANGELOG#L19

we should update the setup.py for this

Reply:
Oh I thought this feature was up since way longer !
Last commit include fixes for `setup.py` (and all other requirements.txts, xonsh-in-docker, ...).
Please check I left no-one behind (worked for me for both `pip -e` and `xonsh-in-docker`) as I'm still new to this tree.

Reply:
It looks good. One thing though, qa checks can be run locally with `xonsh run-tests.xsh qa`